dts/arm/st/stm32l4.dtsi

```diff
@@ -23,21 +23,32 @@
 		};
 	};
 
-	flash0: flash@8000000 {
-		compatible = "soc-nv-flash";
-		label = "FLASH_STM32";
-		reg = <0x08000000 DT_FLASH_SIZE>;
-
-		write-block-size = <8>;
-	};
-
 	sram0: memory@20000000 {
 		device_type = "memory";
 		compatible = "mmio-sram";
 		reg = <0x20000000 DT_SRAM_SIZE>;
 	};
 
 	soc {
+
+		flash-controller@40022000 {
+			compatible = "st,stm32l4-flash-controller";
+			label = "FLASH_CTRL";
+			reg = <0x40022c00 0x400>;
+			interrupts = <4 0>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@8000000 {
+				compatible = "soc-nv-flash";
+				label = "FLASH_STM32";
+				reg = <0x08000000 DT_FLASH_SIZE>;
+
+				write-block-size = <8>;
+			};
```

Comment:
Some L4 can have multiple flash banks, depending on the model.

Reply:
> Some L4 can have multiple flash banks, depending on the model. Maybe the node could be added but left disabled ?

which parts, have multiple banks?

Reply:
Let me find them back... anyway it's not blocking.

Reply:
Some info on Dual bank flash here:
http://www.st.com/content/ccc/resource/training/technical/product_training/91/e3/aa/26/e6/69/4f/de/STM32L4_Memory_Flash.pdf/files/STM32L4_Memory_Flash.pdf/jcr:content/translations/en.STM32L4_Memory_Flash.pdf
So you can address bank 1 or 2 depending on a bit in SYSCFG Memory Rempa register. Not sure how this should appear in device tree